spec/features/bulkrax_dynamic_import_spec.rb

```diff
@@ -0,0 +1,173 @@
+
+      CSV.open(file, "wb", headers: headers, write_headers: true) do |row|
+        number_of_records.times.each do |i|
+          row << ["id-#{i}", "source-#{i}", model_name, depositor.email] + faked_data(i)
+        end
+      end
+
+      file.close
+    end
+
+    def delete_csv_file
+      File.delete(temporary_file_path) if File.exist?(temporary_file_path)
+    end
+
+    def faked_data(i)
+      attributes.collect { |attribute| faked_attribute(attribute.first, i) }
+    end
+
+    # rubocop:disable Metrics/CyclomaticComplexity
+    # rubocop:disable Metrics/PerceivedComplexity
+    # rubocop:disable Metrics/AbcSize
+    def faked_attribute(name, i)
+      return unless name.is_a? Symbol
+
+      if name == :creator_name_type_1
+        "Personal"
+      elsif attributes[name][:type] == "date"
+        Time.zone.at(rand * Time.current.to_i).strftime("%Y-%m-%d")
+      elsif attributes[name][:type] == "select" && attributes.dig(name, :authority)
+        attributes[name][:authority].constantize.new(model: model_name.constantize).select_active_options.sample&.first
+      elsif attributes[name][:type] == "textarea"
+        "A long piece of text. A long piece of text. A long piece of text. A long piece of text. A long piece of text. A long piece of text. A long piece of text. A long piece of text."
+      else
+        "#{name}-#{i}"
+      end
+    end
+    # rubocop:enable Metrics/CyclomaticComplexity
+    # rubocop:enable Metrics/PerceivedComplexity
+    # rubocop:enable Metrics/AbcSize
+
+    # Test methods
+
+    def csv_row_to_expectations(row, work)
+      expectations = []
+
+      field_configs.each do |attribute|
+        attribute_name = attribute.first.to_s
+        attribute_properties = attribute.last
+
+        if work.respond_to?(attribute_name)
+          if attribute_properties.dig(:subfields)
+            # For creator etc, loop through each creator field separately
+            attribute_properties[:subfields].reject { |k, _v| untested_attributes.include?(k) }.each_key do |subfield|
+              expectations << get_expectations_for_subfield(attribute_name, subfield, row, work)
+            end
+          else
+            expectations << get_expectations_for_field(attribute_name, attribute_properties, row, work)
+          end
+        end
+      end
+
+      expectations
+    end
+
+    def get_expectations_for_field(attribute_name, attribute_properties, row, work)
+      attribute_test_value = attribute_properties[:multiple] ? [row[attribute_name]] : row[attribute_name]
+      attribute_actual_value = work.send(attribute_name)
+
+      { attribute_name: attribute_name, actual: attribute_actual_value, test: attribute_test_value }
+    end
+
+    def get_expectations_for_subfield(attribute_name, subfield, row, work)
+      attribute_test_value = row["#{subfield}_1"]
+      attribute_actual_value = JSON.parse(work.send(attribute_name).first, symbolize_names: true).first[subfield]
```

Comment:
Could someone comment on whether there is a better way to do this rather than parsing this field directly.